pkg/providers/goinstall.go

```diff
@@ -0,0 +1,128 @@
+package providers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/apex/log"
+)
+
+type goinstall struct {
+	name, repo, tag, latestURL string
+}
+
+func parseRepo(path string) (string, string, string, string) {
+	repo := path
+	tag := "latest"
+	if i := strings.LastIndex(path, "@"); i > -1 {
+		repo = filepath.Clean(path[:i])
+		tag = path[i+1:]
+	}
+
+	name := path
+	if i := strings.LastIndex(repo, "/"); i > -1 {
+		name = repo[i+1:]
+	}
+
+	latestURL := fmt.Sprintf("https://proxy.golang.org/%s/@latest", repo)
+
+	return repo, tag, name, latestURL
+}
+
+func newGoInstall(repo string) (Provider, error) {
+	repoUrl := strings.TrimPrefix(repo, "goinstall://")
+	repo, tag, name, latestURL := parseRepo(repoUrl)
+	return &goinstall{repo: repo, tag: tag, name: name, latestURL: latestURL}, nil
+}
+
+func (g *goinstall) Fetch(opts *FetchOpts) (*File, error) {
+	goPath, ok := os.LookupEnv("GOPATH")
```

Comment:
how about if we run `go env GOPATH` instead? that should always yield a value given that if `GOPATH` is not set, it'll return `$HOME/go` by default

Reply:
done.